Python Simulator/eventgame.py

```diff
@@ -20,30 +21,95 @@ def __init__(self, players=None, num_players=2):
         self.curr_player_num = len(players)  # the player number who's turn it is
         self.final_round_cnt = -1
         self.game_finished = False
-
-    def start_game(self, piece_num, permutation):
+        self.good_moves = []
+
+    def build_board(self, data):
+        board = Board()
+        for piece in data:
+            if not board.add_piece(player_num=piece["player_num"],
+                        piece_num=piece["piece_num"], permutation_index=piece["permutation"],
+                        coordinates=(piece["x"], piece["y"]), check=False):
+                return (board, 0)
+            self.players[piece["player_num"]-1].remove_piece(piece["piece_num"])
+        return (board, 1)
+
+    def execute_moves(self, moves, add_first_move):
+        results = [2] if add_first_move else []
+        for move in moves:
+            if move["type"]==0:
+                result = self.pass_turn(move["player_num"])
+                results.append(result)
+                if result==-1:
+                    return results
+            else:
+                result = self.move(move["player_num"], move["piece_num"], move["permutation"], move["x"], move["y"])
+                results.append(result)
+                if result==-1:
+                    return results
+        return results
+
+    
+    def load_game(self, game_file):
+        with open(game_file, "r") as game_json:
+            data = json.load(game_json)
+        moves = data["moves"]
+        board = data["board"]
+        if len(board)>0:
+            self.start_game()
+            (board, error) = self.build_board(data["board"])
+            if(error==0):
+                self.started=False
+                return -1
+            self.game_board = board
+            return self.execute_moves(moves, False)
+        else:
+            first_move = moves[0]
+            if self.start_game(first_move["piece_num"], first_move["permutation"])==-1:
+                self.started=False
+                return -1
+            moves = moves[1:]
+            return self.execute_moves(moves, True)
+
+    def store_game(self, game_file):
+        dict = { "board": [], "moves": self.good_moves}
+        with open(game_file, "w") as game_json:
+            json.dump(dict, game_json, indent=2)
+        self.started=False
+                
```

Comment:
Two spaces only between functions.
(PyCharm and similar IDEs should warnings about such things)